integrations/google_genai/src/haystack_integrations/components/generators/google_genai/chat/chat_generator.py

```diff
@@ -0,0 +1,579 @@
+        """
+        callback_name = serialize_callable(self._streaming_callback) if self._streaming_callback else None
+        serialized_tools = serialize_tools_or_toolset(self._tools) if self._tools else None
+        return default_to_dict(
+            self,
+            api_key=self._api_key.to_dict(),
+            model=self._model,
+            generation_config=self._generation_config,
+            safety_settings=self._safety_settings,
+            streaming_callback=callback_name,
+            tools=serialized_tools,
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "GoogleGenAIChatGenerator":
+        """
+        Deserializes the component from a dictionary.
+
+        :param data: Dictionary to deserialize from.
+        :returns: Deserialized component.
+        """
+        deserialize_secrets_inplace(data["init_parameters"], keys=["api_key"])
+        deserialize_tools_or_toolset_inplace(data["init_parameters"], key="tools")
+        init_params = data.get("init_parameters", {})
+        if "streaming_callback" in init_params and init_params["streaming_callback"] is not None:
+            init_params["streaming_callback"] = deserialize_callable(init_params["streaming_callback"])
+        return default_from_dict(cls, data)
+
+    def _handle_streaming_response_direct(
+        self, response_stream, streaming_callback: StreamingCallbackT
+    ) -> Dict[str, Any]:
+        """
+        Handle streaming response from Google Gen AI generate_content_stream.
+
+        :param response_stream: The streaming response from generate_content_stream.
+        :param streaming_callback: The callback function for streaming chunks.
+        :returns: A dictionary with the replies.
+        """
+        try:
+            # Collect all chunks to build the final response
+            all_text_parts = []
+            all_tool_calls = []
+            final_finish_reason = None
+
+            for chunk in response_stream:
+                # Each chunk is a GenerateContentResponse
+                chunk_text = ""
+
+                # Process candidates in the chunk
+                if chunk.candidates:
+                    candidate = chunk.candidates[0]  # Take the first candidate
+
+                    # Extract content from parts
+                    if candidate.content and candidate.content.parts:
+                        for part in candidate.content.parts:
+                            if part.text:
+                                chunk_text += part.text
+                                all_text_parts.append(part.text)
+                            elif part.function_call:
+                                tool_call = ToolCall(
+                                    tool_name=part.function_call.name,
+                                    arguments=dict(part.function_call.args) if part.function_call.args else {},
+                                    id=part.function_call.id,
```

Comment:
This is a specific case here @sjrl because although we create ToolCall with id the `part` coming from Google models doesn't have that value assigned. Google keeps it optional even in their API for now. So we could generate one here conditionally. But not sure if that would make some negative impact.

Reply:
Yup you're right, I've seen the same. The ID field has only ever returned as None for me so far.